main.go

```diff
@@ -212,6 +217,21 @@ func connect(ssl bool, secret string, access string, host string) (*minio.Client
 	return minioClient, err
 }
 
+func (t *bearerToken) authenticate(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		tokenSplit := strings.Split(r.Header.Get("Authorization"), "Bearer")[1]
+		token := strings.TrimSpace(tokenSplit)
+
+		if token == t.token {
```

Comment:
Can you use subtle compare from the crypto package? https://golang.org/pkg/crypto/subtle/#ConstantTimeCompare

Reply:
Done!